src/common/http/post_lambda_request.cpp

```diff
@@ -0,0 +1,31 @@
+#include "post_lambda_request.hpp"
+
+namespace http {
+PostLambda::PostLambda(LambdaT lam, const char *url_string)
+    : hdrs {
+        { nullptr, nullptr, std::nullopt },
+    }
```

Comment:
This is not needed. You're allowed to return `nullptr` from `extra_headers` and it'll do what you expect (even then you wouldn't have to keep it in the instance). In addition, it's what the default implementation does, so even that one is not needed.